test/distributed/org/apache/cassandra/distributed/impl/DistributedTestSnitch.java

```diff
@@ -34,6 +39,29 @@
 public class DistributedTestSnitch extends AbstractNetworkTopologySnitch
 {
     private static NetworkTopology mapping = null;
+    private static final BiMap<InetAddressAndPort, InetSocketAddress> cache = HashBiMap.create();
+
+    static InetAddressAndPort toCassandraInetAddressAndPort(InetSocketAddress addressAndPort)
```

Comment:
I've replaced this with two concurrent hashmaps. Even though access to each map is not syncornised with its inverse, I think it's fine, since the worse case scenario we'll just create one extra object.